WordPress/gutenberg
# [RNMobile] Fix Issue with Search block stealing focus after media updated for image block

**Mobile Gutenberg PR**: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3442




## Description

Fixes #31351 where the Search block was stealing the focus away from the Image block after the image was updated (iOS only). The fix itself was really straightforward - at some point I defaulted the `isPlaceholderSelected` state variable to `true` but this is not necessary and was forcing the search input element of the Search block to gain focus and take focus from the image block. Defaulting this to `false` fixed the issue. While this change was tiny, it required fixes to both the Unit Test snapshot and the e2e tests.  


## How has this been tested?



Tested the following scenario on both Android and iOS:

1. Add a search block
2. Add an image block
3. Select an image to be inserted in the image block
4. Observe that after the image is added the search block gets selected

#### Run unit tests
```
> npm run test-unit:native -- packages/block-library/src/search/test/edit.native.js
```

#### Run e2e tests

**IOS**
```
> npm run native test:e2e:bundle:ios
> TEST_RN_PLATFORM=ios npm run native device-tests:local gutenberg-editor-search.test.js
```

**Android**
```
> npm run native test:e2e:bundle:android
> npm run native device-tests:local gutenberg-editor-search.test.js
```
## Screenshots 

#### Before 
https://user-images.githubusercontent.com/5810477/116763840-f011b380-a9ec-11eb-9530-274dc663ba6a.mov|

#### After

https://user-images.githubusercontent.com/5810477/116763860-03248380-a9ed-11eb-8bd5-a201fe3cfbaf.mov



## Types of changes




Bug fix.

## Checklist:
- [ ] I've tested my changes with keyboard and screen readers. (No changes made that would effect screenreaders)
